libd-llvm/src/d/llvm/type.d

```diff
@@ -310,7 +318,34 @@ final class TypeGen {
 
 		import std.string;
 		auto llvmStruct = typeSymbols[i] = LLVMStructCreateNamed(llvmCtx, i.mangle.toStringz());
+
+		LLVMValueRef[] vtbl = [];
+		foreach (member; i.members) {
+			if (auto m = cast(Method) member) {
+				//vtbl ~= pass.visit(m);
+			}
+		}
+
+		import std.algorithm, std.array;
+		auto vtblTypes = vtbl.map!(m => LLVMTypeOf(m)).array();
 		auto vtblStruct = LLVMStructCreateNamed(llvmCtx, (i.mangle ~ "__vtbl").toStringz());
+		LLVMStructSetBody(vtblStruct, vtblTypes.ptr, cast(uint) vtblTypes.length, false);
+
+		auto vtblPtr = LLVMAddGlobal(dmodule, vtblStruct, (i.mangle ~ "__vtblZ").toStringz());
+		LLVMSetInitializer(vtblPtr, LLVMConstNamedStruct(vtblStruct, vtbl.ptr, cast(uint) vtbl.length));
+		LLVMSetGlobalConstant(vtblPtr, true);
+		LLVMSetLinkage(vtblPtr, LLVMLinkage.LinkOnceODR);
+
+		LLVMValueRef[] fields = [];
+		fields ~= vtblPtr;
+		ivtbls[i] = vtblPtr;
```

Comment:
This is needed. A vtable pointer for the interface is created and later added to the class vtable. The 'fields' might not be needed, which can be removed afterwards.